scripts/check_pre_commit_consistency.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# (C) Copyright Alpine Quantum Technologies GmbH 2024
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# Check consistency between versions of the pre-commit hooks
+# and the packages installed by poetry.
+
+set -euo pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# Retrieve the version of a Python package installed by Poetry.
+installed_package_version() {
+    package_name="$1"
+    # FIXME: install poetry-export plugin to not need to silence the warning here.
+    package_version=$(poetry export \
+           --only dev \
+	   --format requirements.txt \
+	   --without-hashes \
+	   --without-urls 2> /dev/null | \
+	   sed -nr "s/$package_name==([0-9][0-9.]*).*/\1/p")
+    echo "$package_version"
+}
+
+# Retrieve the version of a pre-commit hook.
+pre_commit_hook_version() {
+    tool_name="$1"
+    config_path="$SCRIPT_DIR/../.pre-commit-config.yaml"
+    hook_version=$(yq -r ".repos[] | select(.repo | test(\"$tool_name\")).rev" "$config_path")
+    hook_version=${hook_version#v}
+    echo "$hook_version"
+}
+
+tools=(ruff typos pyproject-fmt interrogate)
+exit_code=0
+
+for tool in "${tools[@]}"; do
+    package=$(installed_package_version "$tool")
+    hook=$(pre_commit_hook_version "$tool")
+    echo -n "$tool: package=$package hook=$hook "
+    if [ -n "$package" ] && [ -n "$hook" ] && [ "$package" = "$hook" ]; then
+	echo " OK"
+    else
+	echo " FAIL"
```

Comment:
Might be nice to print this case to stderr

Reply:
Yeah I meant the whole line. Perhaps print to both in that case, to keep overview?

Reply:
I changed the output format. A successful run looks like:

```
ruff: version=0.7.0 OK
typos: version=1.26.0 OK
pyproject-fmt: version=2.4.3 OK
interrogate: version=1.7.0 OK
```

with all output on `stdout`.

A failed run looks like:

```
ruff: version=0.7.0 OK
typos: version=1.26.0 OK
pyproject-fmt: package=2.4.3 hook=2.4.5 FAIL
interrogate: version=1.7.0 OK
```

with the `FAIL` line on `stderr`.